### PR TITLE
Fix text error bug

### PR DESCRIPTION
**Pull request by bcomnes**
This updates dependencies and adds an error response test.  Down the road it can be adapted to cover the no consumer error, which is caused by a non-json response.

### CODE REVIEW

File: go/porcelain/deploy.go [hunk 1]
```diff
@@ -337,13 +335,6 @@ func (n *Netlify) WaitUntilDeployReady(ctx context.Context, d *models.Deploy, ti
 	return n.waitForState(ctx, d, timeout, "prepared", "ready")
 }
 
-func (n *Netlify) WaitForDeployComplete(ctx context.Context, d *models.Deploy, timeout time.Duration) (*models.Deploy, error) {
-	if timeout <= 0 {
-		timeout = doneProcessingTimeout
-	}
-	return n.waitForState(ctx, d, timeout, "ready")
-}
-
```
Comment (bcomnes) [inline go/porcelain/deploy.go]: Dead code

File: go/porcelain/deploy_test.go [hunk 2]
```diff
@@ -65,3 +76,29 @@ func TestAddWithLargeMedia(t *testing.T) {
...
+	server := httptest.NewServer(http.HandlerFunc(func(rw http.ResponseWriter, req *http.Request) {
+		rw.Header().Set("Content-Type", "application/json; charset=utf-8")
+		rw.WriteHeader(408)
+		rw.Write([]byte(`{ "foo": "bar", "message": "a message", "code": 408 }`))
+	}))
+	defer server.Close()
+
+	httpClient := http.DefaultClient
+	authInfo := runtime.ClientAuthInfoWriterFunc(func(r runtime.ClientRequest, _ strfmt.Registry) error {
+		_ = r.SetHeaderParam("User-Agent", "buildbot")
```
Comment (keiko713) [inline go/porcelain/deploy_test.go]: kinda nitpicking but is this `_ =` part needed?

Comment (bcomnes) [reply go/porcelain/deploy_test.go]: Good question.

Comment (bcomnes) [reply go/porcelain/deploy_test.go]: Nope! Good catch.

**Pull request merged**